consensus/db/db.go

```diff
@@ -60,15 +60,32 @@ func (w *walEntry[V, H, A]) UnmarshalCBOR(data []byte) error {
 }
 
 // TendermintDB defines the methods for interacting with the Tendermint WAL database.
+//
+// The purpose of the WAL is to record any event that may result in a state change.
+// These events fall into the following categories:
+// 1. Incoming messages. We do not need to store outgoing messages.
```

Comment:
We actually do need to store outgoing messages, otherwise the node can double vote.

Reply:
Ah, maybe I get what you mean, because the state machine should be deterministic, so it will always result in the same vote, except for proposal where it calls the `Value()` function which can be non-deterministic.